guacamole/src/main/webapp/app/navigation/services/userPageService.js

```diff
@@ -60,9 +60,13 @@ angular.module('navigation').factory('userPageService', ['$injector',
      * @returns {PageDefinition}
      *     The user's home page.
      */
-    var generateHomePage = function generateHomePage(rootGroups) {
+    var generateHomePage = function generateHomePage(rootGroups,permissions) {
 
         var homePage = null;
+        var settingsPages = generateSettingsPages(permissions);
+
+        if (settingsPages.length > 1)
```

Comment:
I'm generally OK with this check (and honestly can't think of a better way), but please include a comment to document what's going on here. The need for this check is not inherently clear.

Reply:
Commented.